packit_service/worker/reporting.py

```diff
@@ -113,42 +184,92 @@ def report_status_by_comment(
         table = "\n".join(comment_table_rows)
         self.comment(table + f"\n### Description\n\n{description}")
 
-    def __add_commit_comment_with_status(
-        self, state: CommitStatus, description: str, check_name: str, url: str = ""
-    ):
-        body = (
-            "\n".join(
-                [
-                    f"- name: {check_name}",
-                    f"- state: {state.name}",
-                    f"- url: {url if url else 'not provided'}",
-                ]
-            )
-            + f"\n\n{description}"
-        )
-        self.project.commit_comment(
-            commit=self.commit_sha,
-            body=body,
-        )
+    def get_statuses(self):
+        self.project_with_commit.get_commit_statuses(commit=self.commit_sha)
+
+    def comment(self, body: str):
+        if self.pr_id:
+            self.project.get_pr(pr_id=self.pr_id).comment(body=body)
+        else:
+            self.project.commit_comment(commit=self.commit_sha, body=body)
+
+
+class StatusReporterPagure(StatusReporter):
+    @staticmethod
+    def get_commit_status(state: BaseCommitStatus):
+        state = StatusReporter.get_commit_status(state)
+        # Pagure has no running status
+        if state == CommitStatus.running:
+            state = CommitStatus.pending
+
+        return state
 
     def set_status(
         self,
-        state: CommitStatus,
+        state: BaseCommitStatus,
         description: str,
         check_name: str,
         url: str = "",
     ):
+        state_to_set = self.get_commit_status(state)
+        logger.debug(
+            f"Setting Pagure status '{state_to_set.name}' for check '{check_name}': {description}"
+        )
+
         # Required because Pagure API doesn't accept empty url.
-        if not url and isinstance(self.project, PagureProject):
+        if not url:
             url = "https://wiki.centos.org/Manuals/ReleaseNotes/CentOSStream"
 
-        logger.debug(
-            f"Setting status '{state.name}' for check '{check_name}': {description}"
+        self.project_with_commit.set_commit_status(
+            self.commit_sha, state_to_set, url, description, check_name, trim=True
         )
+        # Also set the status of the pull-request for forges which don't do
+        # this automatically based on the flags on the last commit in the PR.
+        self.__set_pull_request_status(check_name, description, url, state_to_set)
+
+    def __set_pull_request_status(
+        self, check_name: str, description: str, url: str, state: CommitStatus
+    ):
```

Comment:
Do you want to get rid of the pagure-specific hacks here or leave it as a follow-up?

(context: https://github.com/packit/packit-service/issues/591#issuecomment-804120367)

Reply:
removed